File: HackerRank Interview Prep Kit/Warm Up/Jumping Clouds.py
def solution():
    n = int(input())
    arr = list(map(int,input().split()))
    count = 0
    idx = 0
    while True:
        if idx + 2 < n and arr[idx+2] == 0:
            idx += 2
            count += 1
        elif idx + 1 < n and arr[idx+1] == 0:
            idx += 1
            count += 1
        else:
            break
    return(count)



if __name__ == '__main__':
    print(solution())

File: HackerRank Interview Prep Kit/Warm Up/jumping_clouds.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    int count = 0;
    int num;
    cin >> num;

    vector<int> arr(num);
    for (int i = 0; i < num; i++)
    {
        cin >> arr[i];
    }

    int i = 0;
    while (i < num)
    {
        if (i + 2 < num && arr[i + 2] == 0)
        {
            count++;
            i += 2;
        }
        else if (i + 1 < num && arr[i + 1] == 0)
        {
            count++;
            i += 1;
        }
        else
        {
            cout << count;
            return 0;
        }
    }

    cout << count;

    return 0;
}